src/QMCWaveFunctions/OrbitalBase.cpp

```diff
@@ -113,11 +113,11 @@ OrbitalBase::RealType OrbitalBase::KECorrection()
   return 0;
 }
 
-void OrbitalBase::get_ratios(ParticleSet& P, std::vector<ValueType>& ratios)
+void OrbitalBase::evaluateRatiosAlltoOne(ParticleSet& P, std::vector<ValueType>& ratios)
 {
-  std::ostringstream o;
-  o << "OrbitalBase::get_ratios is not implemented by " << OrbitalName;
-  APP_ABORT(o.str());
+  assert(P.getTotalNum()==ratios.size());
+  for (int i=0; i<P.getTotalNum(); ++i)
+    ratios[i]=ratio(P,i);
```

Comment:
Is this intended to be a primary implementation or just a (slow) fallback?  If the latter, would it be better to have something available just for testing rather than production (or at least a warning that the code is slow)?

What wavefunction components will currently call this code?

Reply:
In general, this is a fallback solution but it doesn't mean it is slow in all cases.
For determinants, it is slow because it is evaluating SPO nelec times and So we have a specialization to just do once for spin up and once for spin down.
This is the primary implementation for all the Jastrow factors. Since there is little data reuse. You can compare the old implementation in AoS. Since there is no benefit having specialization in Jastrow factors, I removed all the old specialized AoS code for having less code to maintain.